src/api/signals/classification.py

```diff
@@ -0,0 +1,47 @@
+from django.db.models.signals import post_delete, post_save, pre_save
+from django.dispatch import receiver
+
+from ..models import Image
+from ..utils import classification as cls_utils
+
+
+@receiver(pre_save, sender=Image)
+def strip_exif(sender, instance, **kwargs):
+    if not instance.created_on:
+        cls_utils.store_exif(instance)
+        original_image_name = instance.image.name
+        instance.data = instance.data or {}
+        if "original_image_name" not in instance.data:
+            instance.data["original_image_name"] = original_image_name
+
+        image_name = cls_utils.create_unique_image_name(instance)
+        instance.name = image_name
+        instance.image.name = image_name
+
+        cls_utils.correct_image_orientation(instance)
```

Comment:
If any of this fails for any reason, should we prevent the save so that we don't get an image with exif data?